kernel/src/fs/configfs/tsm.rs

```diff
@@ -0,0 +1,206 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use alloc::{boxed::Box, string::ToString, sync::Arc};
+use core::fmt::Debug;
+
+use aster_systree::{
+    inherit_sys_branch_node, inherit_sys_leaf_node, BranchNodeFields, Error, NormalNodeFields,
+    Result, SysAttrSetBuilder, SysObj, SysPerms, SysStr,
+};
+use aster_util::printer::VmPrinter;
+use inherit_methods_macro::inherit_methods;
+use ostd::{
+    mm::{FallibleVmRead, FallibleVmWrite, VmReader, VmWriter},
+    sync::RwLock,
+};
+use spin::Once;
+
+use crate::device::tdxguest::tdx_get_quote;
+
+const TSM_INBLOB_MAX: usize = 64;
+
+#[derive(Debug)]
+struct Tsm {
+    fields: BranchNodeFields<ReportSystem, Self>,
+}
+
+impl Tsm {
+    fn new() -> Arc<Self> {
+        let name = SysStr::from("tsm");
+        let builder = SysAttrSetBuilder::new();
+
+        let attrs = builder.build().expect("Failed to build attribute set");
+
+        Arc::new_cyclic(|weak_self| {
+            let fields = BranchNodeFields::new(name, attrs, weak_self.clone());
+            fields
+                .add_child(ReportSystem::new())
+                .expect("Failed to add report directory");
+
+            Tsm { fields }
+        })
+    }
+}
+
+inherit_sys_branch_node!(Tsm, fields, {
+    fn perms(&self) -> SysPerms {
+        SysPerms::DEFAULT_RW_PERMS
+    }
+});
+
+#[derive(Debug)]
+struct ReportSystem {
```

Comment:
"System" usually refers to a set of entities that have complex interactions. Apparently, the `ReportSystem` here is not up the standard of a system. I suggest renaming it to `ReportSet`.